prime-router/src/main/kotlin/fhirengine/translation/hl7/ValueSetMap.kt

```diff
@@ -0,0 +1,33 @@
+package gov.cdc.prime.router.fhirengine.translation.hl7
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.annotation.JsonTypeInfo
+import java.util.SortedMap
+
+@JsonTypeInfo(
+    use = JsonTypeInfo.Id.DEDUCTION,
+    include = JsonTypeInfo.As.PROPERTY,
+    defaultImpl = InlineValueSet::class
+)
+/**
+ * Interface for valueSet. Implement this interface to create a data provider
+ * that can return a SortedMap<String, String> at runtime.
+ */
+interface ValueSetMap<T> {
+    val configData: T
+    fun getMapValues(): SortedMap<String, String>
```

Comment:
Would it be useful to add `fun getMappedValue(String): String` here? The caller often (or maybe always?) just gets the map just to look up a particular value, so maybe we can simplify usage of the interface with that call.

Reply:
I added this function, moving the logic out of `ConfigSchemaProcessor`.